lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py

```diff
@@ -0,0 +1,85 @@
+"""This module is used to transfer incoming twilio requests to a Vetext endpoint"""
+
+import json
+import http.client
+import os
+import urllib.parse
+import boto3
+import logging
+
+def lambda_handler(event: any, context: any):
+    """this method takes in an event passed in by either an alb or sqs.
+        @param: event   -  contains data pertaining to an incoming sms from Twilio
+        @param: context -  contains information regarding information
+            regarding what triggered the lambda (context.invoked_function_arn).
+    """
+
+    try:
+        assert os.environ.get('vetext_api_endpoint_domain') is not None, 'vetext_api_endpoint_domain'
+        assert os.environ.get('vetext_api_endpoint_auth') is not None, 'vetext_api_endpoint_auth'
+        assert os.environ.get('vetext_api_endpoint_path') is not None, 'vetext_api_endpoint_path'
+        assert os.environ.get('vetext_request_drop_sqs_url') is not None, 'vetext_api_endpoint_path'
+    except AssertionError as e:
+        print(f'Failed to find environmental variable: {e}')
+        # Handle failed env variable
+        return {
+            'statusCode': 424,
+            'body': "Missing env variable"
+        }
```

Comment:
Don't cause an exception to catch it.
```
if 'vetext_api_endpoint_domain' not in os.environ or . . . :
    logger.error(...)
    return {
        ...
    }
```

Reply:
If we want to be "pythonic" we shouldn't even be checking. We should wrap it all in a try/except and catch the key error. Python follows the "[Easier to ask for forgiveness than permission](https://docs.python.org/3.8/glossary.html#term-eafp)" principle. Additionally, we need to check for failures in the http connection, which also need to be wrapped.

Reply:
I concur.